onnxruntime/core/providers/coreml/builders/impl/reshape_op_builder.cc

```diff
@@ -89,39 +118,64 @@ bool ReshapeOpBuilder::IsOpSupportedImpl(const Node& node, const OpBuilderInputP
 
   Initializer unpacked_tensor(*new_shape_tensor);
   auto new_shape = unpacked_tensor.DataAsSpan<int64_t>();
+
   if (new_shape.empty()) {
     LOGS(logger, VERBOSE) << "New shape of reshape cannot be empty";
     return false;
   }
 
+  // CoreML reshape does not support 0 as a dimension
+  // Even though MLProgram Reshape docs imply 0 is supported as a dimension, it is not supported
+  if (allow_zero) {
```

Comment:
nit: isn't this slightly different to the supposedly supported usage of zero (copy value from input)? If allow_zero is set to 1, the ONNX behavior is to copy an explicit zero (which equates to empty output). CoreML doesn't support that at all afaik.